pkg/resourcemanager/controller/secret/reconciler.go

```diff
@@ -66,6 +68,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 			// check if we are responsible for this MR, class might have changed, then we need to remove our finalizer
 			if ref.Name == secret.Name && r.ClassFilter.Responsible(&resource) {
 				secretIsReferenced = true
+				if _, ok := secret.Labels[v1alpha1.ReferencedBy]; !ok {
+					patch := client.MergeFromWithOptions(secret.DeepCopy(), client.MergeFromWithOptimisticLock{})
+					metav1.SetMetaDataLabel(&secret.ObjectMeta, v1alpha1.ReferencedBy, resource.Name)
+					if err := r.SourceClient.Patch(ctx, secret, patch); err != nil {
+						log.Info(fmt.Errorf("failed to add label to secret: %w", err).Error())
```

Comment:
Why you decided to only log the error instead of returning it?

```suggestion
						return reconcile.Result{}, fmt.Errorf("failed to add the %s label to secret: %w", resourcesv1alpha1.ReferencedBy, err)
```